sdk/core/azure-core-http-okhttp/src/main/java/com/azure/core/http/okhttp/implementation/OkHttpAsyncResponse.java

```diff
@@ -111,6 +116,71 @@ public void writeBodyTo(OutputStream outputStream) throws IOException {
         StreamUtils.INSTANCE.transfer(responseBody.byteStream(), outputStream);
     }
 
+    @Override
+    public Mono<Void> writeBodyTo(AsynchronousFileChannel asynchronousFileChannel, long position) {
+        return Mono.defer(
+            () -> {
+                AtomicLong currentPosition = new AtomicLong(position);
+                byte[] buffer = new byte[8 * 1024];
+                ByteBuffer byteBuffer = ByteBuffer.wrap(buffer);
+                BufferedSource source = responseBody.source();
```

Comment:
If we opt to using `FileChannel` here there is this API which may be better suited for this task, as OkHttp's `BufferedSource` implements `ReadableByteChannel`.

https://docs.oracle.com/javase/8/docs/api/java/nio/channels/FileChannel.html#transferFrom-java.nio.channels.ReadableByteChannel-long-long-